app/models/portfolio.rb

```diff
@@ -22,7 +22,9 @@ def add_portfolio_item(portfolio_item)
   end
 
   def metadata
-    {:user_capabilities => user_capabilities,
-     :shared            => self.access_control_entries.any?}
+    {:user_capabilities        => user_capabilities,
+     :groups_shared_count      => self.access_control_entries.collect(&:group_uuid).uniq.count,
```

Comment:
This can be done is SQL with:
`:groups_shared_count      => access_control_entries.select(:group_uuid).distinct.count,`

Why `groups_shared_count` and not `shared_count`?  Is it possible to have it shared multiple times to the same group, and if so what does that actually mean of the end-user?